### PR TITLE
Cpu-strided-complex support for binary-ops

### PR DESCRIPTION
**Pull request by dylanbespalko**
In-tree changes to pytorch to support complex numbers are being submitted here.
Out-of-tree support for complex numbers is here: [pytorch-cpu-strided-complex extension](https://gitlab.com/pytorch-complex/pytorch-cpu-strided-complex)

Note: These changes do not support AVX/SSE operations on complex tensors.
Changes so far:

- [x]  Added complex support of torch.empty.
- [x]  Added complex support of CopyKernels
- [x]  Added complex support of BinaryOp kernels

Once these changes are applied the rest of the kernels are pretty easy.

@ezyang 
I have fixed the issues in the original [PR: 25373](https://github.com/pytorch/pytorch/pull/25373).

### CODE REVIEW

**Comment by dylanbespalko** [comment]
@ezyang 

Do you have time to look at this PR? I'm close to getting the unary ops working. I can add that soon.

**Comment by ezyang** [comment]
Sorry, it got lost in my notifications. Looking.

File: torch/csrc/utils/python_arg_parser.cpp [hunk 1]
```diff
@@ -274,6 +277,9 @@ void FunctionParameter::set_default_str(const std::string& str) {
     default_bool = (str == "True" || str == "true");
   } else if (type_ == ParameterType::DOUBLE) {
     default_double = atof(str.c_str());
+  } else if (type_ == ParameterType::COMPLEX) {
+    default_complex[0] = atof(str.c_str()); // TODO: parse "x + xj"?
```
Comment (ezyang) [inline torch/csrc/utils/python_arg_parser.cpp]: Hehe. Probably not necessary unless we ever do get an imaginary default.

**Review by ezyang — APPROVED** [review]
This is great work, thank you!

**Pull request closed without merging**